### PR TITLE
Bump/tonic and opentelementry

### PR DESCRIPTION
**Pull request by Phoenix500526**
Please briefly answer these questions:
bump versions of tonic, opentelemetry-xxx, etcd-client and madsim
remove some deprecated methods

* what problem are you trying to solve? (or if there's no problem, what's the motivation for this change?)

* what changes does this pull request make?

* are there any non-obvious implications of these changes? (does it break compatibility with previous versions, etc)

### CODE REVIEW

File: crates/utils/src/tracing.rs [hunk 1]
```diff
@@ -89,8 +89,8 @@ mod test {
     };
 
     use super::*;
-    #[test]
-    fn test_inject_and_extract() -> Result<(), Box<dyn std::error::Error>> {
+    #[tokio::test]
+    async fn test_inject_and_extract() -> Result<(), Box<dyn std::error::Error>> {
```
Comment (bsbds) [inline crates/utils/src/tracing.rs:93]: Why this needs to be changed to async?

File: crates/utils/src/tracing.rs [hunk 2]
```diff
@@ -112,8 +112,10 @@ mod test {
 
     /// init tracing subscriber
     fn init() -> Result<(), Box<dyn std::error::Error>> {
-        let jaeger_online_layer = opentelemetry_jaeger::new_agent_pipeline()
-            .with_service_name("test")
+        let otlp_exporter = opentelemetry_otlp::new_exporter().tonic();
+        let jaeger_online_layer = opentelemetry_otlp::new_pipeline()
+            .tracing()
+            .with_exporter(otlp_exporter)
```
Comment (bsbds) [inline crates/utils/src/tracing.rs:118]: Why is this change required? Please also rename `jaeger_online_layer`.

File: crates/xlinectl/Cargo.toml [hunk 3]
```diff
@@ -18,7 +18,8 @@ serde = { version = "1.0.137", features = ["derive"] }
 serde_json = "1.0.107"
 shlex = "1.3.0"
 tokio = "1"
-tonic = "0.10.2"
+# tonic = "0.11.0"
```
Comment (bsbds) [inline crates/xlinectl/Cargo.toml:21]: Please remove the comments

File: crates/xlinectl/src/main.rs [hunk 4]
```diff
@@ -164,7 +164,10 @@ use clap::{arg, value_parser, Command};
 use command::compaction;
 use ext_utils::config::ClientConfig;
 use tokio::fs;
+// #[cfg(not(madsim))]
 use tonic::transport::{Certificate, ClientTlsConfig};
+// #[cfg(madsim)]
+// use utils::ClientTlsConfig;
```
Comment (bsbds) [inline crates/xlinectl/src/main.rs:170]: Please remove the comments.

File: workspace-hack/Cargo.toml [hunk 5]
```diff
@@ -14,45 +14,41 @@ publish = false
...
 futures-channel = { version = "0.3", features = ["sink"] }
 futures-util = { version = "0.3", features = ["channel", "io", "sink"] }
 getrandom = { version = "0.2", default-features = false, features = ["js", "rdrand", "std"] }
-itertools = { version = "0.11" }
 libc = { version = "0.2", features = ["extra_traits"] }
 log = { version = "0.4", default-features = false, features = ["std"] }
-madsim-tokio = { version = "0.2", default-features = false, features = ["fs", "io-util", "macros", "net", "rt", "rt-multi-thread", "signal", "sync", "time"] }
-madsim-tonic = { version = "0.4", default-features = false, features = ["tls"] }
+madsim-tokio = { git = "https://github.com/Phoenix500526/madsim.git", branch = "update-tonic", default-features = false, features = ["fs", "io-util", "macros", "net", "rt", "rt-multi-thread", "signal", "sync", "time"] }
+madsim-tonic = { git = "https://github.com/Phoenix500526/madsim.git", branch = "update-tonic", default-features = false, features = ["tls"] }
```
Comment (bsbds) [inline workspace-hack/Cargo.toml:28]: Please add comments here explaining what are the modifications of this branch.

File: crates/xlinectl/src/main.rs (hunk 4 above)
Comment (Phoenix500526) [reply crates/xlinectl/src/main.rs:170]: Done

File: crates/xlinectl/Cargo.toml (hunk 3 above)
Comment (Phoenix500526) [reply crates/xlinectl/Cargo.toml:21]: Done

File: crates/utils/src/tracing.rs (hunk 2 above)
Comment (Phoenix500526) [reply crates/utils/src/tracing.rs:118]: The opentelemetry_jaeger::new_agent_pipeline is deprecated since 0.21.0. We'd better migrate to opentelemetry-otlp exporter. FYI: https://docs.rs/opentelemetry-jaeger/latest/opentelemetry_jaeger/fn.new_agent_pipeline.html

File: crates/utils/src/tracing.rs (hunk 1 above)
Comment (Phoenix500526) [reply crates/utils/src/tracing.rs:93]: That is because `opentelemetry_otlp` require tokio runtime. Without this modification, this test case will result in "there is no reactor running, must be called from the context of a Tokio 1.x runtime".

File: workspace-hack/Cargo.toml (hunk 5 above)
Comment (Phoenix500526) [reply workspace-hack/Cargo.toml:28]: The `workspace-hack/Cargo.toml` is generated.

Comment (rogercloud) [inline workspace-hack/Cargo.toml:28]: If there's an up-stream issue to track the issue, please comment it here.

**Review by rogercloud — APPROVED** [review]

File: crates/utils/src/tracing.rs (hunk 2 above)
Comment (bsbds) [reply crates/utils/src/tracing.rs:118]: I think it's no long named `jaeger`

Comment (Phoenix500526) [reply crates/utils/src/tracing.rs:118]: But it's indeed use for jaeger. The jaeger is supported otlp now.

**Review by bsbds — APPROVED** [review]
